Docs: Issue with Selective Icon Import from `react-icons` in Next.js

### What is the documentation issue?

### Summary

I followed the instructions from the official Next.js documentation [How to optimize your local development environment#3-check-your-imports](https://nextjs.org/docs/app/guides/local-development#3-check-your-imports) to avoid importing thousands of icons at once. However, the suggested method doesn’t seem to work, and I encountered the following error:


```tsx
// Before
import { MdAttachment, MdOutlineDevices } from "react-icons/md";
```

```tsx
// After
import MdAttachment from "react-icons/md/MdAttachment";
import MdOutlineDevices from "react-icons/md/MdOutlineDevices";
```

Error Messages:

```ts
Cannot find module 'react-icons/md/MdAttachment' or its corresponding type declarations.ts(2307)
Cannot find module 'react-icons/md/MdOutlineDevices' or its corresponding type declarations.ts(2307)
```

### Additional information

I found a similar import pattern in the v2 documentation for `react-icons`: https://github.com/react-icons/react-icons?tab=readme-ov-file#migrating-from-version-2---3

```tsx
// OLD IMPORT STYLE
import FaBeer from "react-icons/lib/fa/beer";

function Question() {
  return (
    <h3>
      Lets go for a <FaBeer />?
    </h3>
  );
}
```

Additionally, the README also mentions another import method using the `@react-icons/all-files` package. However, this seems to be intended for specific use cases (like Meteor.js, Gatsby.js, etc.):
 https://github.com/react-icons/react-icons?tab=readme-ov-file#installation-for-meteorjs-gatsbyjs-etc

```tsx
import { FaBeer } from "@react-icons/all-files/fa/FaBeer";

function Question() {
  return (
    <h3>
      Lets go for a <FaBeer />?
    </h3>
  );
}
```

It’s clear that the documented method is not applicable here. What’s the correct way to import only the icons I need from `react-icons` without importing the entire package?


### Is there any context that might help us understand?

_Originally posted by @1chooo in https://github.com/vercel/next.js/discussions/80799_

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/guides/local-development#3-check-your-imports

--- a/docs/01-app/02-guides/local-development.mdx
+++ b/docs/01-app/02-guides/local-development.mdx
@@ -37,18 +37,19 @@ npm run dev --turbopack
 
 The way you import code can greatly affect compilation and bundling time. Learn more about [optimizing package bundling](/docs/app/guides/package-bundling) and explore tools like [Dependency Cruiser](https://github.com/sverweij/dependency-cruiser) or [Madge](https://github.com/pahen/madge).
 
-### Icon libraries
+#### Icon libraries
 
-Libraries like `@material-ui/icons` or `react-icons` can import thousands of icons, even if you only use a few. Try to import only the icons you need:
+Libraries like `@material-ui/icons`, `@phosphor-icons/react`, or `react-icons` can import thousands of icons, even if you only use a few. Try to import only the icons you need:
 
 ```jsx
 // Instead of this:
-import { Icon1, Icon2 } from 'react-icons/md'
+import { TriangleIcon } from '@phosphor-icons/react'
 
 // Do this:
-import Icon1 from 'react-icons/md/Icon1'
-import Icon2 from 'react-icons/md/Icon2'
+import { TriangleIcon } from '@phosphor-icons/react/dist/csr/Triangle'
 ```
+
+You can often find what import pattern to use in the documentation for the icon library you're using. This example follows [`@phosphor-icons/react`](https://www.npmjs.com/package/@phosphor-icons/react#import-performance-optimization) recommendation.
 
 Libraries like `react-icons` includes many different icon sets. Choose one set and stick with that set.
 
@@ -61,13 +62,13 @@ For example, if your application uses `react-icons` and imports all of these:
 
 Combined they will be tens of thousands of modules that the compiler has to handle, even if you only use a single import from each.
 
-### Barrel files
+#### Barrel files
 
 "Barrel files" are files that export many items from other files. They can slow down builds because the compiler has to parse them to find if there are side-effects in the module scope by using the import.
 
 Try to import directly from specific files when possible. [Learn more about barrel files](https://vercel.com/blog/how-we-optimized-package-imports-in-next-js) and the built-in optimizations in Next.js.
 
-### Optimize package imports
+#### Optimize package imports
 
 Next.js can automatically optimize imports for certain packages. If you are using packages that utilize barrel files, add them to your `next.config.js`:
 

